tests/material/test_immiscibleflash.cpp

```diff
@@ -61,38 +62,30 @@ void checkSame(const FluidState& fsRef, const FluidState& fsFlash)
 
         // check the pressures
         error = 1 - fsRef.pressure(phaseIdx)/fsFlash.pressure(phaseIdx);
-        if (std::abs(error) > tol) {
-                std::ostringstream oss;
-                oss << "pressure error phase " << phaseIdx << " is incorrect: "
-                    << fsFlash.pressure(phaseIdx)  << " flash vs "
-                    << fsRef.pressure(phaseIdx) << " reference"
-                    << " error=" << error;
-                throw std::runtime_error(oss.str());
-        }
+        BOOST_CHECK_MESSAGE(std::abs(error) <= tol,
+                            "pressure error phase " + std::to_string(phaseIdx) +
+                            " is incorrect: " + std::to_string(fsFlash.pressure(phaseIdx)) +
+                            " flash vs " + std::to_string(fsRef.pressure(phaseIdx)) +
+                            " reference error=" + std::to_string(error));
```

Comment:
My memory is hazy on the details, but I **vaguely** recall `BOOST_CHECK_MESSAGE` supporting streaming operators for string-like objects.  Are you able to write something like
```C++
BOOST_CHECK_MESSAGE(std::abs(error) <= tol,
"pressure error phase " << phaseIdx << " is incorrect: " << fsFlash.pressure(phaseIdx)
<< " flash vs " << fsRef.pressure(phaseIdx) << " reference error = " << error);
```
here?  Possibly at a cost of raising the minimum Boost version?

Reply:
indeed. updated.